### PR TITLE
streamlookup: Fix UAF when GetAsyncInputData called before Bootstrap or after PassAway

### PR DESCRIPTION
**Pull request by yumkam**
### Changelog entry 

...

### Changelog category 

* Bugfix 
* Not for changelog (changelog entry is not required)

### Additional information

[v3]
Setting `InputFlowFetchStatus` to `Finished` implicitly prevents `KeysForRequests` dereferences after `PassAway`.
Get rid of `TActorBootstrapped`, it works unreliable with direct calls to `GetAsyncInputData`.
[v4]
There were use-before-initialization MaxKeysForRequest (used for initial size of KeysForRequest hashtable)

### CODE REVIEW

File: ydb/library/yql/dq/actors/input_transforms/dq_input_transform_lookup.cpp [hunk 1]
```diff
@@ -217,15 +200,45 @@ class TInputTransformStreamLookupBase
...
         Y_UNUSED(freeSpace);
         auto startCycleCount = GetCycleCountFast();
         auto guard = BindAllocator();
 
         DrainReadyQueue(batch);
 
-        if (InputFlowFetchStatus != NUdf::EFetchStatus::Finish && KeysForLookup->empty()) {
+        if (InputFlowFetchStatus != NUdf::EFetchStatus::Finish && GetKeysForLookup()->empty()) {
             Y_DEBUG_ABORT_UNLESS(AwaitingQueue.empty());
+            GetLookupSourceId();
```
Comment (dorooleg) [inline ydb/library/yql/dq/actors/input_transforms/dq_input_transform_lookup.cpp:241]: Тут игнорируется значение. Давай назовем функцию PrepareLookupSourceId которая не будет ничего возвращать

File: ydb/library/yql/dq/actors/input_transforms/dq_input_transform_lookup.cpp [hunk 2]
```diff
@@ -217,15 +200,44 @@ class TInputTransformStreamLookupBase
...
         Y_UNUSED(freeSpace);
         auto startCycleCount = GetCycleCountFast();
         auto guard = BindAllocator();
 
         DrainReadyQueue(batch);
 
-        if (InputFlowFetchStatus != NUdf::EFetchStatus::Finish && KeysForLookup->empty()) {
+        if (InputFlowFetchStatus != NUdf::EFetchStatus::Finish && GetKeysForLookup()->empty()) {
             Y_DEBUG_ABORT_UNLESS(AwaitingQueue.empty());
+            PrepareLookupSourceId();
```
Comment (kardymonds) [inline ydb/library/yql/dq/actors/input_transforms/dq_input_transform_lookup.cpp:240]: А нельзя для простоты перенести в самое начало GetAsyncInputData() / почему оно именно под if?

**Review by kardymonds — DISMISSED** [review]

File: ydb/library/yql/dq/actors/input_transforms/dq_input_transform_lookup.cpp (hunk 2 above)
Comment (yumkam) [reply ydb/library/yql/dq/actors/input_transforms/dq_input_transform_lookup.cpp:240]: Подкладывание соломки: если почему-то GetAsyncInputData был вызван после PassAway, то тут ничего не будет вызвано

**Comment by yumkam** [comment]
There are transient/flapping errors in streamlookup tests (`test_join.py.*Streaming`) in v3 (that moved initialization from `Bootstrap` to `GetAsyncInputData`). They are not reported as hard errors (those tests are muted), but usually they work, so needs more testing locally.

**Review by dorooleg — APPROVED** [review]

**Review by rvu1024 — APPROVED** [review]

**Pull request merged**